### ISSUE TITLE
dragon and myrmex lag in version 1.7.1-1.12.2

### ISSUE THREAD

**Issue by kreezxil** [Alex-the-666/Ice_and_Fire#1510]
see also #1331 which i have closed because that one was resolved.

However this lag appears to be coming in from the onlivingupdate. 

Dragons are lagging hard as are Myrmex, but the sea serpents are behaving.

![dragon 10kms lag](https://user-images.githubusercontent.com/3880072/55436360-b34f0600-5561-11e9-9b08-846467bddc30.png)
![myrmex 6 8kms lag](https://user-images.githubusercontent.com/3880072/55436369-b944e700-5561-11e9-9f40-24ef2cf31bd6.png)
![serpent 204ms no lag](https://user-images.githubusercontent.com/3880072/55436374-bd710480-5561-11e9-86e8-321c90847fde.png)

spark data: https://sparkprofiler.github.io/#jZcIlIt4x6

**Comment by suken27** [comment]
It seems the AI of the dragons and the myrmex takes lots of execution time.

**Comment by Thethuthinnang** [comment]
I am having this issue, as well, with this release.

**Comment by kreezxil** [comment]
That confirms it for me, I thought I was having a one-off, I'll be rolling
my copy of I&F down I believe 2 versions which is where it wasn't having
the issue.

**Comment by Thethuthinnang** [comment]
Yeah, I've just gone back to 1.6.1. I was enjoying all the fixes and additions in the latest version, but the lag made it effectively unplayable on my system. (I'll miss those sea serpents. They were boss.)

**Comment by kreezxil** [comment]
I loved, nay relished hearing my players talk of the horrors of the sea.
Ear plugs satisfied the Sirens, but nothing save your blood would save them
from sea serpents.

**Comment by kreezxil** [comment]
I'm going to close this because now after more than 4 different profiling tools we have come to the conclusion that players with greedy got have it all, it must run all the time, let's figure out how to do a chunkloader despite the restrictions were causing so much server lag that your mod probably got more blame than it deserved.

**Comment by kreezxil** [comment]
I'm reopening it because we've been managing player caused lag and this is creeping back up again. Definitely Myrmex related lag. 

![image](https://user-images.githubusercontent.com/3880072/60907530-67d28500-a23f-11e9-8460-f57d0f72cc41.png)

spark report: https://sparkprofiler.github.io/#e2lS7pYTED

other server staff I recommend this awesome profiling tool https://github.com/lucko/spark

**Comment by kreezxil** [comment]
in other news I think this report is related https://github.com/Alex-the-666/Ice_and_Fire/issues/1531 but I don't think their solution is the right way to go either.

**Comment by kreezxil** [comment]
I rebuilt my modpack for World of Dragons and we're experiencing mega lag again. Decided to run the spark profiler for 4 hours and here are results. Lots of **MYRMEX** issues in the following.

# visual report 1
![Screenshot_1](https://user-images.githubusercontent.com/3880072/61314671-c7420f00-a7c2-11e9-9783-8dfa32c81154.png)

# visual report 2
![Screenshot_3](https://user-images.githubusercontent.com/3880072/61314683-cb6e2c80-a7c2-11e9-9a19-3145eb5503e7.png)

# visual report 3
![Screenshot_4](https://user-images.githubusercontent.com/3880072/61314693-cf9a4a00-a7c2-11e9-8528-080eec8536b2.png)

# visual report 4
![Screenshot_5](https://user-images.githubusercontent.com/3880072/61314700-d32dd100-a7c2-11e9-857b-e0b87227c874.png)

# visual report 5
![Screenshot_6](https://user-images.githubusercontent.com/3880072/61314706-d5902b00-a7c2-11e9-910f-d43f0d930027.png)

# raw spark data 
Be warned it's huge and will slow your browser down while you try to poke thru it
https://sparkprofiler.github.io/#BZqBeADEqh

@Alex-the-666 what's the ETA on the next version? is this lag issue fixed in it?

**Comment by AlexModGuy** [comment]
![Screenshot (155)](https://user-images.githubusercontent.com/8662109/62729210-00aa1b00-ba0d-11e9-9a71-9e2d911fa6fc.png)
@kreezxil I've got all the myrmex using up ~30% of the server resources down to ~4%. keep in mind that this is in a world with a shit ton of myrmex and deathworms constantly fighting.

Pretty much, I made their AI a bit stupider and optimized the worker code for looking for cocoons to store items in.

The update should be in a few days!
